src/ml/neural_net/style_transfer/mps_style_transfer_utils.m

```diff
@@ -0,0 +1,657 @@
+
+  vgg16Desc.block3.conv1.kernelWidth = 3;
+  vgg16Desc.block3.conv1.kernelHeight = 3;
+  vgg16Desc.block3.conv1.inputFeatureChannels = 128;
+  vgg16Desc.block3.conv1.outputFeatureChannels = 256;
+  vgg16Desc.block3.conv1.strideWidth = 1;
+  vgg16Desc.block3.conv1.strideHeight = 1;
+  vgg16Desc.block3.conv1.paddingWidth = 1;
+  vgg16Desc.block3.conv1.paddingHeight = 1;
+  vgg16Desc.block3.conv1.label = @"vgg16_block_3_conv_1";
+  vgg16Desc.block3.conv1.updateWeights = NO;
+
+  vgg16Desc.block3.conv2.kernelWidth = 3;
+  vgg16Desc.block3.conv2.kernelHeight = 3;
+  vgg16Desc.block3.conv2.inputFeatureChannels = 256;
+  vgg16Desc.block3.conv2.outputFeatureChannels = 256;
+  vgg16Desc.block3.conv2.strideWidth = 1;
+  vgg16Desc.block3.conv2.strideHeight = 1;
+  vgg16Desc.block3.conv2.paddingWidth = 1;
+  vgg16Desc.block3.conv2.paddingHeight = 1;
+  vgg16Desc.block3.conv2.label = @"vgg16_block_3_conv_2";
+  vgg16Desc.block3.conv2.updateWeights = NO;
+
+  vgg16Desc.block3.conv3.kernelWidth = 3;
+  vgg16Desc.block3.conv3.kernelHeight = 3;
+  vgg16Desc.block3.conv3.inputFeatureChannels = 256;
+  vgg16Desc.block3.conv3.outputFeatureChannels = 256;
+  vgg16Desc.block3.conv3.strideWidth = 1;
+  vgg16Desc.block3.conv3.strideHeight = 1;
+  vgg16Desc.block3.conv3.paddingWidth = 1;
+  vgg16Desc.block3.conv3.paddingHeight = 1;
+  vgg16Desc.block3.conv3.label = @"vgg16_block_3_conv_3";
+  vgg16Desc.block3.conv3.updateWeights = NO;
+
+  vgg16Desc.block3.pooling.kernelSize = 2;
+  vgg16Desc.block3.pooling.strideSize = 2;
+
+  vgg16Desc.block4.conv1.kernelWidth = 3;
+  vgg16Desc.block4.conv1.kernelHeight = 3;
+  vgg16Desc.block4.conv1.inputFeatureChannels = 256;
+  vgg16Desc.block4.conv1.outputFeatureChannels = 512;
+  vgg16Desc.block4.conv1.strideWidth = 1;
+  vgg16Desc.block4.conv1.strideHeight = 1;
+  vgg16Desc.block4.conv1.paddingWidth = 1;
+  vgg16Desc.block4.conv1.paddingHeight = 1;
+  vgg16Desc.block4.conv1.label = @"vgg16_block_4_conv_1";
+  vgg16Desc.block4.conv1.updateWeights = NO;
+
+  vgg16Desc.block4.conv2.kernelWidth = 3;
+  vgg16Desc.block4.conv2.kernelHeight = 3;
+  vgg16Desc.block4.conv2.inputFeatureChannels = 256;
+  vgg16Desc.block4.conv2.outputFeatureChannels = 512;
+  vgg16Desc.block4.conv2.strideWidth = 1;
+  vgg16Desc.block4.conv2.strideHeight = 1;
+  vgg16Desc.block4.conv2.paddingWidth = 1;
+  vgg16Desc.block4.conv2.paddingHeight = 1;
+  vgg16Desc.block4.conv2.label = @"vgg16_block_4_conv_2";
+  vgg16Desc.block4.conv2.updateWeights = NO;
+
+  vgg16Desc.block4.conv3.kernelWidth = 3;
+  vgg16Desc.block4.conv3.kernelHeight = 3;
+  vgg16Desc.block4.conv3.inputFeatureChannels = 256;
+  vgg16Desc.block4.conv3.outputFeatureChannels = 512;
+  vgg16Desc.block4.conv3.strideWidth = 1;
+  vgg16Desc.block4.conv3.strideHeight = 1;
+  vgg16Desc.block4.conv3.paddingWidth = 1;
+  vgg16Desc.block4.conv3.paddingHeight = 1;
+  vgg16Desc.block4.conv3.label = @"vgg16_block_4_conv_3";
+  vgg16Desc.block4.conv3.updateWeights = NO;
```

Comment:
I see what you mean with your offline comment about this being verbose. FWIW, in hindsight I think we could have done something more programmatic, looping over block index and conv index, if these were stored as arrays somehow instead of named components. See, for example, how the Python OD toolkits defines the tiny darknet model: https://github.com/apple/turicreate/blob/master/src/python/turicreate/toolkits/object_detector/_model.py

Not anything we need to change now, just something to consider for future reference

Reply:
Alright, I'll keep that in mind when doing future implementations in MPS!